redo of COA recent collabs builder

hufngvuowng opened the pull request:
closes #300

Commit:
test file

Commit:
added coa_template.xlsx

Commit:
- added script coabuilder.py filling in excel template
- working on fixing cells with data validation

hufngvuowng commented:
sorry for the late push. I was having some trouble running regolith this morning. Several cells has data validation, which isn't removed when i added new rows in, so currently it's looking like this. I'm gonna work on removing the data validation.

![image](https://user-images.githubusercontent.com/40038591/74977224-528a1f80-53f8-11ea-8af8-e70c5f5924e5.png)

sbillinge commented:
exciting!  Thanks.  I will review a little later.

S

On Thu, Feb 20, 2020 at 3:47 PM Hung Vuong <notifications@github.com> wrote:

> sorry for the late push. I was having some trouble running regolith this
> morning. Several cells has data validation, which isn't removed when i
> added new rows in, so currently it's looking like this. I'm gonna work on
> removing the data validation.
>
> [image: image]
> <https://user-images.githubusercontent.com/40038591/74977224-528a1f80-53f8-11ea-8af8-e70c5f5924e5.png>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sbillinge/regolith/pull/16?email_source=notifications&email_token=ABAOWUP6DUYV7IJ6DU5K563RD3T65A5CNFSM4KYABXEKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMQAHXI#issuecomment-589300701>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAOWUJGB4HEJ2EUFOUK7RTRD3T65ANCNFSM4KYABXEA>
> .
>


-- 
Professor Simon Billinge
Columbia University

sbillinge commented on regolith/builders/coabuilder.py:
```diff
@@ -0,0 +1,173 @@
+    def excel(self):
+        rc = self.rc
+        gtx = self.gtx
+        since_date = dt.date.today() - relativedelta(months=48)
```
let's remove the hard-coded 48

hufngvuowng replied:
Are we going to declare a global variable at the top like NMONTHS = 48 and then pass that into the excel function?

sbillinge reviewed (commented):
Thanks @tienhungvuong this is excellent!  A couple of comments below:
1) we need to remove duplicates.  The easiest way I have found is ``ppl = list(set(ppl))`` (I think ppl is better variable name than ppl_names because it also contains their institution).
2) I think it should be last name, first name, middle initial.    This should be easily done using the properties of the HumanName object, so ``ppl[0].last`` or sthgl like that will return the last name, etc..  
3) ideally the formatting will go pale blue, paler blue, pale blue paler blue etc., but it may be more trouble than it is worth.  I won't lose funding because the cells are the wrong color.....  Fix it if it is straightforward.
4) Later we will have to filter and remove people who were former post-docs and students and mentors etc, and put them in different tables, but we will have to refactor the schema a bit for that to work and it can be done later.   If you can basically fix the formatting we can merge so that people can be hacking on this version.

This is super!

sbillinge commented:
Oh, I guess one more thing...I think the entries have to be in alphabetic order by last name, so we will need a line that sorts by last name.

sbillinge commented:
Hi @tienhungvuong I think I want to merge this.  Do you have anything you need to push before I do that?

hufngvuowng commented:
Hi @sbillinge I'm reading through your comments and suggestions now. Do you want me to work on them first before merging?

Commit:
- removed the duplicate entries
- reformatted the names into Last, First format
- sorted the names by alphabetical order (by last name)

sbillinge commented:
That’s right

On Fri, Feb 21, 2020 at 10:30 AM Hung Vuong <notifications@github.com>
wrote:

> *@tienhungvuong* commented on this pull request.
> ------------------------------
>
> In regolith/builders/coabuilder.py:
>
> > +            reverse=True,
> +        )
> +        gtx["contacts"] = sorted(
> +            all_docs_from_collection(rc.client, "contacts"),
> +            key=position_key,
> +            reverse=True,
> +        )
> +        gtx["institutions"] = all_docs_from_collection(rc.client,
> +                                                       "institutions")
> +        gtx["citations"] = all_docs_from_collection(rc.client, "citations")
> +        gtx["all_docs_from_collection"] = all_docs_from_collection
> +
> +    def excel(self):
> +        rc = self.rc
> +        gtx = self.gtx
> +        since_date = dt.date.today() - relativedelta(months=48)
>
> Are we going to declare a global variable at the top like NMONTHS = 48 and
> then pass that into the excel function?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sbillinge/regolith/pull/16?email_source=notifications&email_token=ABAOWULTBT7IFIVRQ65LIADRD7XPTA5CNFSM4KYABXEKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCWPISFA#discussion_r382644899>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAOWUMZX6KQBLPXOWIIM6TRD7XPTANCNFSM4KYABXEA>
> .
>
-- 
Professor Simon Billinge
Columbia University

Commit:
added global variable NUM_MONTHS

hufngvuowng commented:
> That’s right
> On Fri, Feb 21, 2020 at 10:30 AM Hung Vuong ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In regolith/builders/coabuilder.py: > + reverse=True, + ) + gtx["contacts"] = sorted( + all_docs_from_collection(rc.client, "contacts"), + key=position_key, + reverse=True, + ) + gtx["institutions"] = all_docs_from_collection(rc.client, + "institutions") + gtx["citations"] = all_docs_from_collection(rc.client, "citations") + gtx["all_docs_from_collection"] = all_docs_from_collection + + def excel(self): + rc = self.rc + gtx = self.gtx + since_date = dt.date.today() - relativedelta(months=48) Are we going to declare a global variable at the top like NMONTHS = 48 and then pass that into the excel function? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <#16?email_source=notifications&email_token=ABAOWULTBT7IFIVRQ65LIADRD7XPTA5CNFSM4KYABXEKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCWPISFA#discussion_r382644899>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABAOWUMZX6KQBLPXOWIIM6TRD7XPTANCNFSM4KYABXEA> .
> -- Professor Simon Billinge Columbia University

sounds good. I added that just now. Also I can't seem to figure out how to overwrite the existing data validation. I guess worst case we can just remove the validation manually after filling in the excel file.

sbillinge commented:
thanks @tienhungvuong I merged.  Please synchronize your local and kill this branch.  Make a new branch for future commits, but now others can work on this version for the next steps too.

hufngvuowng commented:
Sounds good! Thank you!
